### PR TITLE
Upgraded Makefile

### PR DESCRIPTION
**Pull request by criadoperez**
Improvements on Makefile:

* Checks dependencies for `go`, `docker` and `curl` when using commands that require them
* Changes detection of architecture from `arch` to `uname -a` which is generally more portable and widely available on a broader number of systems.

### CODE REVIEW

**Review by christophercampbell — CHANGES_REQUESTED** [review]
Make this work for the `build-docker*` targets

**Comment by criadoperez** [comment]
Thanks for the comments @christophercampbell . 
In that case it's best to separate the checks. I'll commit the new code with this change.

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,33 @@
...
+ifndef CHECK_DOCKER
+	$(error "Docker is not installed. Please install Docker and retry.")
+endif
+
+# Targets that require the checks
+build: check-go check-curl
+build-docker: check-go check-curl check-docker
+build-docker-nc: check-go check-curl check-docker
+lint: check-go check-curl check-docker
+install-linter: check-go check-curl
```
Comment (christophercampbell) [inline Makefile:28]: These seem over determined. Only install-linter needs curl, for example

**Review by christophercampbell — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (criadoperez) [reply Makefile]: Thanks for pointing this out. Corrected them to the minimum requirements.

**Review by christophercampbell — APPROVED** [review]

**Pull request merged**